flocker/provision/_aws.py

```diff
@@ -24,37 +28,40 @@ def provision_aws(node, package_source, distribution, variants):
         'fedora-20': 'fedora',
         'centos-7': 'centos',
     }[distribution]
-    run(
+
+    commands = []
+
+    commands.append(run_remotely(
         username=username,
         address=node.address,
         commands=task_install_ssh_key(),
-    )
+    ))
 
+    pre_reboot_commands = []
```

Comment:
This works because for the current supported OSs there is one and only one reboot in the provision process for AWS. If that is a safe assumption for the future, fine. If that is something that you think will not cause too much hassle if it isn't the case then fine, but at least document that here.

This has also brought up that the docstring is sparse and missing parameter documentation, so please add that.